### DISCUSSION TITLE
Does WiX v4 support Database deployment and/or DacPac installation?

### DISCUSSION THREAD

**Discussion by nathan-rogers-US9** [wixtoolset/issues#7873] [Questions]
I'm trying to build a WiX installer to set up everything my application needs in one step. This includes copying project output files (which I figured out), setting up a database, installing a service, and eventually setting up IIS and/or Kestrel browser. 

I've noticed that a lot of the documentation around v4 is relatively new, which is to be expected, but I was wondering how fleshed out the WiX/Heat implementation for v4 actually is. Does it support the same features that v3 did? Is there documentation or a place I can find assistance with these items?

Thanks

**Comment by chrpai** [comment]
I have a FOSS project that has project templates and graphical designers to handle the IIS and Windows service stories without needing Heat.  As for dacpac deployment, it's technically possible to wire this into your installer but it's really complicated.  I generaly reccomend this to be pushed to a first run / post install configuration step.   This way you can write the UX, business rules and script execution in your native language that you are more familiar with.

**Reply by nathan-rogers-US9** [reply]
So just to clarify, you are suggesting that I should have the dacpac deployment be part of the program itself and run on first run/post install right? Not part of a script/action within WiX?

**Reply by chrpai** [reply]
Correct.  For an example of UX install Azure DevOps Server on a VM.  They have a nice simple install and then use an admin wizard to configure everything including creating/updating databases, creating service accounts and configuring IIS.

It's not that WiX/MSI can't do these things... it just takes a whole lot of expertise and lots of development time and I personally find it easier to do and a better user experience when done outside/after the installation.

I once did a project for an state government application where they had a requirement that development hand everything off to IT and they do the deployment.  I implmented it this way and they loved it.
